Src/DependencyCollector/Shared/Implementation/HttpDesktopDiagnosticSourceSubscriber.cs

```diff
@@ -48,6 +48,7 @@ public void OnNext(DiagnosticListener value)
                 if (value.Name == "System.Net.Http.Desktop")
                 {
                     this.sourceSubscription = value.Subscribe(this.parent, (Predicate<string>)null);
+                    DependencyTableStore.Instance.IsDesktopHttpDiagnosticSourceActivated = true;
```

Comment:
We are counting on HttpDesktopDiagnosticSourceSubscriber OnNext() to set IsDesktopHttpDiagnosticSourceActivated = true. We definitely need unit test to check this.

Also, is there a guarantee that OnNext will be called before any callbacks to Framework event source?? If not, the IsDesktopHttpDiagnosticSourceActivated will be False, and FrameworkEventSource will start collecting dependencies, which might be later collected by DiagnosticSource. (this may be artificial scenario, but i'd like to know if thats even possible)